python/cudf/cudf/core/multiindex.py

```diff
@@ -765,7 +765,20 @@ def _values(self):
     def _concat(cls, objs):
         from cudf import DataFrame, MultiIndex
 
-        source_data = [o._source_data for o in objs]
+        source_data = []
+        for o in objs:
+            if not isinstance(o, MultiIndex):
+                raise TypeError(
+                    "all objects should be of type "
+                    "MultiIndex for MultiIndex concatenation"
+                )
+            source_data.append(o._source_data)
```

Comment:
Can we do this in a list comprehension instead?

Reply:
Split the check into the caller method